src/gui/MainWindow.cpp

```diff
@@ -38,33 +38,24 @@ void MainWindow::selectFile()
         QStandardPaths::writableLocation(QStandardPaths::DesktopLocation),
         tr("Config File (*.json)"));
 
-    if (filename.isNull() || filename.isEmpty()) {
-        qDebug() << "filename not valid";
-        return;
-    }
-
-    ConfigParser parser;
-    m_config.reset(new Configuration{ parser.parse(filename) });
-
-    if (m_config.isNull()) {
-        qWarning() << "parsing error";
-    }
-
-    m_serialConnect->setEnabled(true);
-    m_serialSettings->setEnabled(true);
-
-    emit importFinished({});
+    importConfig(filename);
 }
 
 void MainWindow::createConfigView()
 {
+    auto kids = this->findChildren<QDockWidget *>("ConfigView");
+    if (!kids.isEmpty()) {
+        qDeleteAll(kids);
+    }
+
     auto dock = ConfigViewFactory().makeConfigView(m_config->protocol);
+    dock->setObjectName("ConfigView");
     dock->setParent(this);
     this->addDockWidget(Qt::DockWidgetArea::LeftDockWidgetArea, dock,
                         Qt::Orientation::Vertical);
 }
 
-void MainWindow::create_menubar()
+void MainWindow::createMenubar()
```

Comment:
please align to camelCase createMenuBar

Reply:
I'm aligned! This commet is inopportune

Reply:
The Item is a Menu Bar so to be compliant to camel case you should write menuBar, in your case it's createMenuBar

Please refer to those links:
https://en.wikipedia.org/wiki/Menu_bar
https://en.wikipedia.org/wiki/Camel_case